### PR TITLE
Fixes #66

### LINKED ISSUE #66: 🪲[Bug]: Seamless User Experience: Eliminating SplashScreen Return : Handle System back button
Users should never return to SplashScreen.

**Screenshots**

https://github.com/SB2318/UltimateHealth/assets/87614560/ba33aeb0-42c1-493c-bcdc-1b848c8f88d8

### PR DESCRIPTION
**Pull request by sidharth2829**
#### PR Description
Please include a summary of the changes and the related issue. Describe your changes in detail.

#### Type of Change
- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Documentation update

#### Select your work-area
- [x] Frontend
- [ ] Backend
- [ ] Documentation
- [ ] Others

#### Related Issue
Please provide a link to the issue solved if applicable.
https://github.com/SB2318/UltimateHealth/issues/66

#### Add your Work Example
📷 Add a Snapshot
Screenshot 2024-05-27 at 3 59 21 PM

#### Fixes (mention the issue number which this fixes)
#66 
#closes

#### Checklist
- [x] I have updated my branch and synced it with the project's 'develop' branch before making this PR.
- [x] I have optimized the file changes.
- [x] I have added a snapshot of my work example.
- [x] I have made a PR to the project's develop branch.

#### Undertaking
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] I have checked for plagiarism and assure its authenticity.
- [x] I have read and followed the code of conduct for this repository. I understand that violation of this undertaking may have legal consequences.

- [x] I Agree

### CODE REVIEW

**Review by SB2318 — COMMENTED** [review]
Nice work! please attach the recording of the respective devices.

**Comment by sidharth2829** [comment]
https://github.com/SB2318/UltimateHealth/assets/111569459/caf60f42-510c-4cbb-98d5-66523fea2684

As you can see i can't swipe back to splash screen as swipegesture is disabled on navigationtab.

**Review by SB2318 — CHANGES_REQUESTED** [review]
In Android devices, it does not work.


https://github.com/SB2318/UltimateHealth/assets/87614560/c2fc18b9-56e2-4d74-b5db-fa6e6f789f71

Please fix the bug and share the recording of both devices **(Android and IOS )** here.

**Comment by SB2318** [comment]
> Screen.Recording.2024-05-27.at.4.49.11.PM.mov 
> As you can see i can't swipe back to splash screen as swipegesture is disabled on navigationtab.

This is not done in correct way.
Go to  TabScreens,
Then with swipe, app must be **shut down.**

**Comment by sidharth2829** [comment]
> > Screen.Recording.2024-05-27.at.4.49.11.PM.mov
> > As you can see i can't swipe back to splash screen as swipegesture is disabled on navigationtab.
> 
> This is not done in correct way. Go to TabScreens, Then with swipe, app must be **shut down.**

ok, then i will build this on android studio and make the requested changes.

**Comment by SB2318** [comment]
@sidharth2829 any update?

**Comment by sidharth2829** [comment]
> @sidharth2829 any update?

Yes, i am still trying to launch it on android studio. Since, this stack is completely new to me and also i am working on MacOS, so it's a bit complicated on it.

File: frontend/src/navigations/StackNavigation.js [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
+import { createStackNavigator } from '@react-navigation/stack';
+import { BackHandler, Alert } from 'react-native';
+import { useFocusEffect } from '@react-navigation/native';
+import TabNavigation from './TabNavigation';
 import SplashScreen from '../screens/SplashScreen';
 
 const Stack = createStackNavigator();
 
-const StackNavigation = () => {
+const StackNavigation = ({ navigation, route }) => {
```
Comment (SB2318) [inline frontend/src/navigations/StackNavigation.js:10]: @sidharth2829  your problem lies in this line, see here you passing  ({navigation, route}) at the function definition, but you haven't changed your App. tsx file, where you call this function There it looks like.

image

You are not sending the props there.

In that case, My suggestion is to call the **StackNavigation** component properly,

or you can use  **useRoute** hook from 'react-navigation/native'

code example:

import { useRoute } from '@react-navigation/native';

```
const StackNavigation =  ()=>{
  const route = useRoute();

  // Access route properties
  console.log(route.params);

  return (
    // rest of your code
  );
}

```

Comment (sidharth2829) [reply frontend/src/navigations/StackNavigation.js:10]: > @sidharth2829 your problem lies in this line, see here you passing ({navigation, route}) at the function definition, but you haven't changed your App. tsx file, where you call this function There it looks like.
> 
> image
> 
> You are not sending the props there.
> 
> In that case, My suggestion is to call the **StackNavigation** component properly,
> 
> or you can use **useRoute** hook from 'react-navigation/native'
> 
> code example:
> 
> import { useRoute } from '@react-navigation/native';
> 
> ```
> const StackNavigation =  ()=>{
>   const route = useRoute();
> 
>   // Access route properties
>   console.log(route.params);
> 
>   return (
>     // rest of your code
>   );
> }
> ```

Ok, i will make appropriate changes and i have messaged you on discord. Can you help to run it on android as well?

Comment (SB2318) [reply frontend/src/navigations/StackNavigation.js:10]: I haven't receive, but it will resolve your issue for android also.

**Comment by sidharth2829** [comment]
susmita, i have tried changing the export path but again and again new issues arise. Sometimes it is sdkmanager, sometimes java version and keeps continuing. Can you pls unassign me this task.

**Comment by SB2318** [comment]
> susmita, i have tried changing the export path but again and again new issues arise. Sometimes it is sdkmanager, sometimes java version and keeps continuing. Can you pls unassign me this task.

You are almost done with the job and already spend too much time in it.

**Comment by SB2318** [comment]
please update your branch, and don't add local changes as of now.

**Comment by sidharth2829** [comment]
> please update your branch, and don't add local changes as of now.

ok

**Comment by sidharth2829** [comment]
> please update your branch, and don't add local changes as of now.



> > susmita, i have tried changing the export path but again and again new issues arise. Sometimes it is sdkmanager, sometimes java version and keeps continuing. Can you pls unassign me this task.
> 
> You are almost done with the job and already spend too much time in it.

i know and i personally learned a lot by working on this project, but i think i need more knowledge regarding app development before working on them.

**Comment by SB2318** [comment]
> > please update your branch, and don't add local changes as of now.
> 
> > > susmita, i have tried changing the export path but again and again new issues arise. Sometimes it is sdkmanager, sometimes java version and keeps continuing. Can you pls unassign me this task.
> > 
> > 
> > You are almost done with the job and already spend too much time in it.
> 
> i know and i personally learned a lot by working on this project, but i think i need more knowledge regarding app development before working on them.

see, this is system issue. you have done with your job. But still, if u want to resolve those issues, u can take your time. and contact me on discord.

**Comment by sidharth2829** [comment]
> > > please update your branch, and don't add local changes as of now.
> > 
> > 
> > > > susmita, i have tried changing the export path but again and again new issues arise. Sometimes it is sdkmanager, sometimes java version and keeps continuing. Can you pls unassign me this task.
> > > 
> > > 
> > > You are almost done with the job and already spend too much time in it.
> > 
> > 
> > i know and i personally learned a lot by working on this project, but i think i need more knowledge regarding app development before working on them.
> 
> see, this is system issue. you have done with your job. But still, if u want to resolve those issues, u can take your time. and contact me on discord.

ok i will try to resolve them once more

**Comment by SB2318** [comment]
> > > > please update your branch, and don't add local changes as of now.
> > > 
> > > 
> > > > > susmita, i have tried changing the export path but again and again new issues arise. Sometimes it is sdkmanager, sometimes java version and keeps continuing. Can you pls unassign me this task.
> > > > 
> > > > 
> > > > You are almost done with the job and already spend too much time in it.
> > > 
> > > 
> > > i know and i personally learned a lot by working on this project, but i think i need more knowledge regarding app development before working on them.
> > 
> > 
> > see, this is system issue. you have done with your job. But still, if u want to resolve those issues, u can take your time. and contact me on discord.
> 
> ok i will try to resolve them once more

Good. Then first take a break,and forget about the issue, will connect after 3-4 hours.

**Pull request closed without merging**